src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -286,6 +285,24 @@ public FeatureProvider getProvider(String domain) {
         return providerRepository.getProvider(domain);
     }
 
+    /**
+     * Return the state of the default provider.
+     */
+    public ProviderState getProviderState() {
+        return providerRepository.getProviderState();
+    }
+
+    /**
+     * Get the state of the provider for a domain. If no provider with the domain is found, returns the state of the
+     * default provider.
+     *
+     * @param domain The domain to look for.
+     * @return A named {@link FeatureProvider}
+     */
+    public ProviderState getProviderState(String domain) {
```

Comment:
@chrfwow did this but I had to revert it. We had multiple providers in the contribs who used an encapsulation strategy and used this publicly instead of in a subclass.

Reply:
@guidobrei sorry, I put this comment in the wrong place. I meant for it to be about making the emit protected, which we had to revert (https://github.com/open-feature/java-sdk/pull/1096/files/cd56f25cbfb918117ec53a69ec1d55a87ca6422b#r1758249819)

I think you're right about this one. I don't think these need to be public.